### PR TITLE
Create modal changelog item

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

For https://github.com/uswds/uswds/pull/5315

## Preview link

Preview link:
[Modal changelog →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-modal-refactor-changelog/components/modal/#latest-updates)

### CODE REVIEW

File: _data/changelogs/component-modal.yml [hunk 1]
```diff
@@ -2,6 +2,14 @@ title: Modal
 type: component
 changelogURL:
 items:
+- date: NNNN-NN-NN
```
Comment (mejiaj) [inline _data/changelogs/component-modal.yml:5]: Noting that we'll need to add date before merge.

File: _data/changelogs/component-modal.yml [hunk 2]
```diff
@@ -2,6 +2,14 @@ title: Modal
...
+  summary: Refactors Modal javascript for added consistency and failsafe measures. 
+  summaryAdditional: This resolves issues when initialization is called more than once on a document.
+  isBreaking: false
+  affectsJavascript: true
+  githubPr: 5315
+  githubRepo: uswds
+  versionUswds: 3.5.0
```
Comment (mejiaj) [inline _data/changelogs/component-modal.yml:12]: Need to update version before merge.

**Review by amyleadem — CHANGES_REQUESTED** [review]
Generally looks good. I added a couple of small suggestions in the comments:

File: _data/changelogs/component-modal.yml [hunk 3]
```diff
@@ -2,6 +2,14 @@ title: Modal
 type: component
 changelogURL:
 items:
+- date: NNNN-NN-NN
+  summary: Refactors Modal javascript for added consistency and failsafe measures. 
+  summaryAdditional: This resolves issues when initialization is called more than once on a document.
```
[the remark this answers is not in the document]
Comment (mahoneycm) [reply _data/changelogs/component-modal.yml]: To me, the main "change/improvement" revolves around the failsafe measures, and the JS multi-initialization prevention is a result of that. What do you think of this change? 

```yml
  summary: Updated JavaScript to include fallbacks and error messages when initialization cannot be completed.
  summaryAdditional: Additionally, this improved code consistency and prevented errors when trying to initialize modal twice.
```

Or would it be better practice to address the problem statement first like your example?

[the remark this answers is not in the document]
Comment (amyleadem) [reply _data/changelogs/component-modal.yml]: I like the wording on that update, @mahoneycm. Happy to defer to you about identifying the primary benefit here.

**Review by amyleadem — APPROVED** [review]

**Pull request merged**